File: app/demo/src/main/java/net/gotev/recycleradapterdemo/leavebehind/LeaveBehindViewHolder.java
package net.gotev.recycleradapterdemo.leavebehind;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;

import net.gotev.recycleradapter.RecyclerAdapterNotifier;
import net.gotev.recycleradapter.RecyclerAdapterViewHolder;
import net.gotev.recycleradapterdemo.R;

/**
 * Base RecyclerAdapterViewHolder to extend when implementing leave-behind material pattern.
 * @author Aleksandar Gotev
 */

public abstract class LeaveBehindViewHolder extends RecyclerAdapterViewHolder {

    FrameLayout contentView;
    FrameLayout leaveBehindView;

    public LeaveBehindViewHolder(final View itemView, final RecyclerAdapterNotifier adapter) {
        super(itemView, adapter);

        Context context = itemView.getContext();

        contentView = (FrameLayout) findViewById(R.id.swipe_content_view);
        leaveBehindView = (FrameLayout) findViewById(R.id.swipe_background_layout);

        LayoutInflater.from(context).inflate(getContentViewId(), contentView);
        LayoutInflater.from(context).inflate(getLeaveBehindId(), leaveBehindView);
    }

    public abstract int getContentViewId();

    public abstract int getLeaveBehindId();

}

File: app/demo/src/main/java/net/gotev/recycleradapterdemo/leavebehind/LeaveBehindViewHolder.kt
package net.gotev.recycleradapterdemo.leavebehind

import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_leave_behind.*
import net.gotev.recycleradapter.RecyclerAdapterNotifier
import net.gotev.recycleradapter.RecyclerAdapterViewHolder

/**
 * Base RecyclerAdapterViewHolder to extend when implementing leave-behind material pattern.
 * @author Aleksandar Gotev
 */

abstract class LeaveBehindViewHolder(itemView: View, adapter: RecyclerAdapterNotifier)
    : RecyclerAdapterViewHolder(itemView, adapter), LayoutContainer {

    override val containerView: View?
        get() = itemView

    private val contentView: FrameLayout by lazy { swipe_content_view }
    private val leaveBehindView: FrameLayout by lazy { swipe_background_layout }

    abstract val contentViewId: Int
    abstract val leaveBehindId: Int

    init {
        LayoutInflater.from(itemView.context).apply {
            inflate(contentViewId, contentView)
            inflate(leaveBehindId, leaveBehindView)
        }
    }

}
